### DISCUSSION TITLE
Contributions with plots.contributions_pie()

### DISCUSSION THREAD

**Discussion by lauramenezess** [pastas/pastas#805] [Q&A] [answered]
Hello!
Do you have any advices of when to choose _partition="std"_ or _partition="sum"_ while plotting the contributions? Considering that I have pumping in my model I am not sure of how to handle it when calculating the contributions and which method would be more appropriate. 
Thanks!

Laura

**Comment by martinvonk** [comment]
Hi @lauramenezess,

Please note that we deprecated the contributions_pie plot: https://github.com/pastas/pastas/issues/711 because: 
> @raoulcollenteur: _it is not the best representation of the contribution because these are correlated._

In the paper from Collenteur et al. (2023): [Analysis of nationwide groundwater monitoring networks using lumped-parameter models](https://doi.org/10.1016/j.jhydrol.2023.130120) there is an alternative method proposed based on the _relative importance_. Maybe that can help you out. The methods are fully reproducible and available on zenodo: https://doi.org/10.5281/zenodo.7494481.

**Reply by dbrakenhoff** [reply, marked as answer]
See also discussion #793 which is partly related to your question. Quoting from my own response there:

> **Quantifying contributions**
> As for your question with regards to quantifying the contribution, this is challenging. One method is to compute the variance of each contribution and compare those values. However, with pumping, the variance of the contribution of the wells is highly dependent on the time period you select for your analysis. If the pump was already on, the variance of the contribution might be relatively small while the total drawdown might be significant. If you want to look at the absolute effect of each contribution and express it as a single statistic, how do you deal with negative and positive contributions? Another option is to compute the improvement of the model fit as you add stresses to it. This gives you some statistic of the relative importance of each stress. See for example [this paper by Collenteur et al. (2023)](https://doi-org.tudelft.idm.oclc.org/10.1016/j.jhydrol.2023.130120) (section 4.4). A final, simple option, that doesn't require too much thinking, might be to just plot the contribution of each stress over time in a single plot.

**Reply by lauramenezess** [reply]
Hello @martinvonk and @dbrakenhoff,
Thanks a lot for your answers.
I took the code available on zenodo and I tried to reproduce the method used in Collenteur et al. (2023) but it got complicated with 3 stresses (recharge, river, pumping)...
The objective for me is to compare the contributions between different zones. Since I don't have much time left to spend on it, I will try to discuss it visually using a plot for the three contributions for each zone as suggested by @dbrakenhoff. 

Cheers,
Laura

**Reply by dbrakenhoff** [reply]
Yeah, i can imagine things get complicated with 3 stresses... 

If you can, feel free to upload the figure you are now using for your analysis. It could be useful for others :)! 

Depending on the time period you're analyzing it might still be possible to reduce time-plots of the contributions to a single statistic, but that really depends on your situation and data, and those time plots will help determine what might be possible.
